### ISSUE TITLE
Adjust minimap size for viewing_range

### ISSUE THREAD

**Issue by lhofhansl** [minetest/minetest#10582]
Latest Minetest built from master.

The minimap always have a fixed size: 256, 128, 64 for the map, 512, 256, 128 for the radar.
That is the length of the size of a square, not the radius, which is 1/2 that. That all is fine as long as the viewing_range is typically 100, but if you increase that, then suddenly you look further than the map or even the radar.

We should calibrate the size with the viewing_range.

**Comment by hecktest** [comment]
I'd rather freeze any efforts on the minimap in favor of SPS which should just make the current map hud obsolete.
The minimap is obscenely slow at the moment and increasing its radius might make it even slower.

**Comment by lhofhansl** [comment]
Fair enough, the code looks convoluted. I don't know what SPS is, though.

**Comment by rubenwardy** [comment]
Server-Provided Scripts. It's a proposed new name for SSCSM

**Comment by lhofhansl** [comment]
Looks like this ain't happening any time soon. Reopening, and I'll probably do a trivial PR.

**Comment by v-rob** [comment]
> I'd rather freeze any efforts on the minimap in favor of SPS

Even with #10801, the minimap is still effectively impossible to do from Lua, and of course SSCSM doesn't exist yet.  A follow up PR to that one could make the minimap doable from Lua, and then the minimap can be implemented in the new UI, but until then (which is who knows when), I see no problem with improving the minimap as it is.

**Comment by lhofhansl** [comment]
Over a year later. No other proposed solutions. I'll file a PR with my (hardcoded) suggestion.
I said it many times before: I take a working imperfect solution over the perfect one that does not exist any time.
